runtime/dart_isolate.cc

```diff
@@ -198,11 +198,18 @@ std::weak_ptr<DartIsolate> DartIsolate::CreateRunningRootIsolate(
     root_isolate_create_callback();
   }
 
+#if !OS_FUCHSIA
+  const std::vector<std::string>& args = dart_entrypoint_args;
+#else
```

Comment:
I think this isn't quite right. @arbeng will know, but I think Fuchsia has some code paths that use the embedder API now, so we may nee to duplicate some code in these blocks.

Reply:
I also realized it, and I changed it back. Is the new implementation detail reasonable for you?

Reply:
I don't think you need to do this actually.  The Fuchsia embedder is still in Github and doesn't use Embedder API yet.

So, you can just call run_configuration.SetEntryppintArgs here:  https://github.com/flutter/engine/blob/main/shell/platform/fuchsia/flutter/engine.cc#L354

You'll have to pass the dart_entrypoint_args into the fuchsia Engine, rather than putting them in the settings:
https://github.com/flutter/engine/blob/main/shell/platform/fuchsia/flutter/component_v1.cc#L127

Then you can get rid of these #if OS_FUCHSIA entirely.

We have integration test in CQ, so you should be able to land this without building/testing fuchsia yourself.

Reply:
> I don't think you need to do this actually. The Fuchsia embedder is still in Github and doesn't use Embedder API yet.
> 
> So, you can just call run_configuration.SetEntryppintArgs here: https://github.com/flutter/engine/blob/main/shell/platform/fuchsia/flutter/engine.cc#L354
> 
> You'll have to pass the dart_entrypoint_args into the fuchsia Engine, rather than putting them in the settings: https://github.com/flutter/engine/blob/main/shell/platform/fuchsia/flutter/component_v1.cc#L127
> 
> Then you can get rid of these #if OS_FUCHSIA entirely.
> 
> We have integration test in CQ, so you should be able to land this without building/testing fuchsia yourself.


Thanks for your help!  I'll do that.

Reply:
Done